### DISCUSSION TITLE
Wildcard filtering

### DISCUSSION THREAD

**Discussion by candcconsulting** [iTwin/iTwinUI-react#907] [Q&A] [answered]
I was looking to add a wildcard filter ... something like 
`filterMethod : (filter, row) => {
 rule = "^" + filter.value + "$"
 const regex = new Regexp(rule)
 regex.text(row.[filter.id])}`
 is filterMethod enabled ?
 How would you recommend adding wildcards ?
 Note the above does not "enable" wildcards ... just regular expression filtering
 Also we would like to find blanks ... how do you recommend that is done with the current filtering ?

**Comment by mayank99** [comment, marked as answer]
I'm assuming you're asking about our `Table` component. You can pass a function to the `filter` field in the column.

```ts
{
  id: "my-coiumn",
  Header: "My column",
  accessor: "my-column",
  fieldType: "text",
  Filter: tableFilters.TextFilter(),
  filter: customFilterFn
},
```
```ts
const customFilterFn = React.useCallback((rows, columnIds, filterValue) => {
  // do regex here and return the filtered array
  return rows;
}, []);
```

For more info, check out react-table-v7 docs: https://react-table-v7.tanstack.com/docs/api/useFilters#usefilters

**Reply by candcconsulting** [reply]
Thanks ... that did indeed work ```

  const customFilterFn = (rows: any, columnId: string[], filterString : string) => {

    const regExp = ( filterString.replace('*', '.*'))
    const filteredRows = rows.filter((o : any) => o.values[columnId[0]].match(regExp, 'i'))
    if (filteredRows && filteredRows.length > 0)
      return filteredRows;
      return [];


  }
```
